src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_PatternSwitchStatement.cs

```diff
@@ -391,24 +360,101 @@ private void LowerBasicSwitch(DecisionTree.ByValue byValue)
                 }
```

Comment:
Assert that the special type is not `boolean`? (assuming that all boolean lowering should go through the other method).